tests/cli/test_mila_init_ssh_config.py

```diff
@@ -0,0 +1,349 @@
+        """,
+        # NOTE: Not using this test case because `ssh_config` will generate the outputs with the
+        # same indent, which will make our `expected_blocks` not match the actual output.
+        # """\
+        # # a comment
+        # Host foo
+        #     HostName foobar.com
+        # """,
+        """\
+        # a comment
+        Host foo
+          HostName foobar.com
+
+        # another comment
+        """,
+    ],
+)
+def test_mila_init_no_existing_entries(
+    initial_contents: str | None,
+    accept_mila: bool,
+    accept_mila_cpu: bool,
+    accept_mila_gpu: bool,
+    accept_mila_computenode: bool,
+    confirm_changes: bool,
+    tmp_path: Path,
+):
+    """Checks what entries are added to the ssh config file when running the corresponding portion
+    of `mila init`.
+    """
+    # TODO: This doesn't completely work with the `questionary` package yet.
+    expected_blocks = []
+    expected_blocks += [expected_block_mila] if accept_mila else []
+    expected_blocks += [expected_block_mila_cpu] if accept_mila_cpu else []
+    expected_blocks += [expected_block_mila_gpu] if accept_mila_gpu else []
+    expected_blocks += [expected_block_compute_node] if accept_mila_computenode else []
+
+    ssh_config_path = tmp_path / ".ssh" / "config"
+    ssh_config_path.parent.mkdir(parents=True, exist_ok=False)
+
+    if initial_contents:
+        initial_contents = textwrap.dedent(initial_contents)
+
+    if initial_contents is not None:
+        with open(ssh_config_path, "w") as f:
+            f.write(initial_contents)
+
+    user_inputs = [
+        "bob\r",  # username
+        _yn(accept_mila),
+        _yn(accept_mila_cpu),
+        _yn(accept_mila_gpu),
+        _yn(accept_mila_computenode),
+        _yn(confirm_changes),
+    ]
+    if not confirm_changes:
+        expected_contents = initial_contents
+    elif initial_contents:
+        # If there is already something in the config file, the new entries should be appended at
+        # the end.
+        # TODO: Confirm this with @Olexa
+        expected_contents = initial_contents + _join_blocks(*expected_blocks)
+    else:
+        expected_contents = _join_blocks(*expected_blocks)
+
+    with set_user_inputs(user_inputs) as input_pipe:
+        if not any([accept_mila, accept_mila_cpu, accept_mila_gpu, accept_mila_computenode]):
+            # Won't get prompted for confirmation if no changes are made.
+            should_exit = False
+        else:
+            should_exit = not confirm_changes
+        with (pytest.raises(SystemExit) if should_exit else contextlib.nullcontext()):
+            setup_ssh_config_interactive(ssh_config_path=ssh_config_path, _input=input_pipe)
+
+    # NOTE: this will stay None if the file wasn't created.
+    resulting_contents: str | None = None
+    if ssh_config_path.exists():
+        with open(ssh_config_path) as f:
+            resulting_contents = f.read()
+
+    assert resulting_contents == expected_contents
```

Comment:
```suggestion
    file_regression.check(resulting_contents)
```

with `pytest-regression`?

Reply:
Fixed in 14f005c